test/helpers/kubectl.go

```diff
@@ -476,19 +476,23 @@ func (kub *Kubectl) WaitForKubeDNSEntry(name string) error {
 	}
 	// https://bugs.launchpad.net/ubuntu/+source/bind9/+bug/854705
 	digCMD := "dig +short %s @%s | grep -v -e '^$'"
+	digCMDFallback := "dig +tcp %s @%s"
 
-	podsIPs, err := kub.GetPodsIPs(KubeSystemNamespace, "k8s-app=kube-dns")
+	host, _, err := kub.GetServiceHostPort(KubeSystemNamespace, "kube-dns")
```

Comment:
in theory this works like the old loop on pidIPs but it might be racey if the kube-dns backends are breakage looping. It's ok anyway, but it might show up as flakiness since it will fail random tests.

Reply:
Not sure if I understand this one. At the moment we're getting the service IP (Pods are using that service IP) so we don't need to loop over dns pods anymore. We're using the service.

Reply:
For some reason I thought we were getting one of the backend IPs for the service, not the service's IP. This is actually more correct, as you say. Ignore me!